Module_3A/hang-man_game.rb

```diff
@@ -0,0 +1,93 @@
+# Putting things into class:
+
+class Game
+    attr_reader :lives,:board,:my_secret_word
+    # -------INIT STATE------------
+    def initialize(word_List)
+        @lives = 7
+        @my_secret_word = word_List.sample
+        @board = set_up_the_board
+    end
+
+    # ---------SETTING UP THE BOARD---------
+    def set_up_the_board
+        Array.new(my_secret_word.size){"_"}
+    end
+
+
+    # ---------BOARD STATE----------
+    def board_state
+        board.join(" ")
+    end
+
+    # --------WIN STATUS------------
+    def won?
+        board.join("") == my_secret_word
+    end
+
+    def lost?
+        lives <= 0
+    end
+
+    # -------- MAKE GUESS -----------
+    def make_guess
+        print "Please guess a letter:"
+        gets.chomp #implicit return occurs here
+    end
+
+    def update_board(guess)
+        if my_secret_word.include?(guess)
+            i = 0
+            while i < my_secret_word.size
```

Comment:
Same as previous example, check if we can avoid looping here

Reply:
Updated :+1: